File: src/main/java/com/vjiki/music/dto/BandResponse.java
package com.vjiki.music.dto;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BandResponse {
    private UUID id;
    private String name;

    @JsonProperty("sort_name")
    private String sortName;

    @JsonProperty("country_code")
    private String countryCode;

    @JsonProperty("is_band")
    private Boolean isBand;

    @JsonProperty("debut_year")
    private Short debutYear;

    private Integer popularity;

    @JsonProperty("created_at")
    private LocalDateTime createdAt;

    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;

    @JsonProperty("cover_url")
    private String coverUrl;

    @Builder.Default
    private List<SongResponse> songs = new ArrayList<>();
}

File: src/main/kotlin/com/vjiki/music/dto/BandResponse.kt
package com.vjiki.music.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime
import java.util.UUID

data class BandResponse(
    val id: UUID,
    val name: String,
    @JsonProperty("sort_name")
    val sortName: String,
    @JsonProperty("country_code")
    val countryCode: String?,
    @JsonProperty("is_band")
    val isBand: Boolean,
    @JsonProperty("debut_year")
    val debutYear: Short?,
    val popularity: Int,
    @JsonProperty("created_at")
    val createdAt: LocalDateTime?,
    @JsonProperty("updated_at")
    val updatedAt: LocalDateTime?,
    @JsonProperty("cover_url")
    val coverUrl: String?,
    val songs: List<SongResponse> = emptyList()
)


